types/unused-webpack-plugin/index.d.ts

```diff
@@ -0,0 +1,52 @@
+// Type definitions for unused-webpack-plugin 2.4
+// Project: https://github.com/MatthieuLemoine/unused-webpack-plugin#readme
+// Definitions by: Remco Haszing <https://github.com/remcohaszing>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+import { WebpackPluginInstance, Compiler } from 'webpack';
+
+export = UnusedWebpackPlugin;
+
+declare namespace UnusedWebpackPlugin {
+    interface UnusedWebpackPluginOptions {
+        /**
+         * array of directories where to look for unused source files.
+         */
+        directories: string[];
+
+        /**
+         * array of exclude patterns when looking for unused source files.
+         */
+        exclude?: string[];
+
+        /**
+         * Root directory that will be use to display relative paths instead of absolute ones
+         */
+        root?: string;
```

Comment:
this field is required 
https://github.com/MatthieuLemoine/unused-webpack-plugin/blob/master/index.js#L8

Reply:
Although the constructor doesn’t assign a default value, the readme explicitly shows that the root is optional https://github.com/MatthieuLemoine/unused-webpack-plugin#options

Reply:
good. so all options are optional